eng/scripts/live-test-resource-cleanup.ps1

```diff
@@ -337,7 +337,8 @@ function DeleteArmDeployments([object]$ResourceGroup) {
   if (!$DeleteArmDeployments) {
     return
   }
-  $toDelete = @(Get-AzResourceGroupDeployment -ResourceGroupName $ResourceGroup.ResourceGroupName | Where-Object { $_ -and $_.Outputs?.Count })
+  $toDelete = @(Get-AzResourceGroupDeployment -ResourceGroupName $ResourceGroup.ResourceGroupName `
+              | Where-Object { $_ -and ($_.Outputs?.Count -or $_.Parameters?.ContainsKey('testApplicationSecret')) })
```

Comment:
Should we just use `contains("secret")`

Reply:
I think this will actually false positive a bunch of stuff, as it appears some deployments have the empty object which will match on the substring. I could check the value of all keys matching `*secrets*` but it seems excessive.

```
secrets              SecureObject               null
```

Reply:
It seems odd to special case just this one parameter name is there something special about this specific one?

Reply:
This is the one we pass in from new-testresources.ps1 so by convention a lot of live test templates use it. Otherwise I only have seen secrets in some azure function deploys from azd.